SwanCustomEnvironments/swancustomenvironments/scripts/makenv.sh

```diff
@@ -0,0 +1,292 @@
+#!/bin/bash
+
+# Author: Rodrigo Sobral 2024
+# Copyright CERN
+# This script allows to create a venv virtual environment to use in notebooks and terminals. The environment contains the packages from a provided requirements file.
+
+# By default, we use the Python installed in the image
+ACCPY_BASE=$ACCPY_PATH/base
+ACCPY_ALL_VERSIONS_STR="?"
+PYTHON_PATH=$(which python)
+if [ -d "$ACCPY_PATH" ]; then
+    ACCPY_ALL_VERSIONS=$(ls -tr $ACCPY_BASE)
+    ACCPY_ALL_VERSIONS_STR=$(echo $ACCPY_ALL_VERSIONS | tr ' ' ', ')
+fi
+
+_log () {
+    if [ "$*" == "ERROR:"* ] || [ "$*" == "WARNING:"* ] || [ "${JUPYTER_DOCKER_STACKS_QUIET}" == "" ]; then
+        echo "$@"
+    fi
+}
+
+# Function for printing the help page
+print_help() {
+    _log "Usage: makenv --env/-e NAME --req/-r REQUIREMENTS [--accpy ACCPY_VERSION] [--clear/-c] [--help/-h]"
+    _log "Options:"
+    _log "  -e, --env NAME              Name of the custom virtual environment (mandatory)"
+    _log "  -r, --req REQUIREMENTS      Path to requirements.txt file or http link for a public repository (mandatory)"
+    _log "  -c, --clear                 Clear the current virtual environment, if it exists"
+    _log "  -h, --help                  Print this help page"
+    _log "  --accpy VERSION             Version of Acc-Py to be used (options: ${ACCPY_ALL_VERSIONS_STR})"
+}
+
+# --------------------------------------------------------------------------------------------
+
+# Parse command line arguments
+while [ $# -gt 0 ]; do
+    key="$1"
+    case $key in
+        --env|-e)
+            ENV_NAME=$2
+            shift
+            shift
+            ;;
+        --req|-r)
```

Comment:
Is this the path to the requirements file or you want to make it the repository that contains the requirements file to match what the users sets in the web form?

Reply:
For consistency with the UI, I would change it to `--repository`, and it can be both an EOS path or a git repo URL. Then, inside both you look for the requirements.txt. This should also simplify a bit your logic when parsing this argument.